src/r_tipping/RateSystem.jl

```diff
@@ -0,0 +1,98 @@
+# we consider the ODE dxₜ/dt = f(xₜ,λ(rt))
+# here λ = λ(t) ∈ Rᵐ is a function containing all the system parameters 
+
+# We ask the user to define: 
+#  1) a ContinuousTimeDynamicalSystem that should be investigated and
+#  2) a protocol for the time-dependent forcing with the struct RateProtocol
+
+# Then we give back the ContinuousTimeDynamicalSystem with the parameter 
+# changing according to the rate protocol
+"""
+    RateProtocol
+
+The RateProtocol contains all the fields (information) that allows to take an ODE of the form 
+    `dxₜ/dt = f(xₜ, λ)` 
+with `λ ∈ Rᵐ` containing all system parameters, and make it an ODE of the form 
+    `dxₜ/dt = f(xₜ, λ(rt))`
+with `λ(t) ∈ Rᵐ` constant before time `t_start` and also after time `t_end`. 
+
+RateProtocol contains the following fields:
+- `λ::Function`: forcing function of the form `λ(p, t_start; kwargs...)``
```

Comment:
We have a couple of issues here. First, it is stated wrongly. It is `λ(p, t; kw...)` it is not a function of the starting time, it is a function of _time_ in general. You can see this clearly in the source code. Second, what is the purpose of `kwargs`? The user makes the function themselves. They don't need `kwargs`. If they want to capture variables inside this function they can make a closure. I think `kwargs` unecessarily complicates this, especially because it is not discussed at all in the docstring here.

Also, is `p` supposed to be `p_lambda` from below? If yes, why not give the same name? (1 word per concept rule)